File: in_java/Pattern_4.java
package in_java;

public class Pattern_4 {
    int row, column;
    String symbol;

    public Pattern_4(int row, int column, String symbol) {
        this.row = row;
        this.column = column;
        this.symbol = symbol;
    }

    public void print() {
        for (int i = 0; i < this.row; i++) {
            for (int j = 0; j < this.column; j++) {
                if (j < row - i) {
                    System.out.print(this.symbol);
                } else {
                    System.out.print(" ");
                }
            }
            System.out.println();
        }
    }
}

File: in_rust/src/pattern_4.rs
pub fn pattern_4() {
    let row = 5;
    let column = 5;
    let symbol = "*";

    for i in 0..row {
        for j in 0..column {
            if j < row - i {
                print!("{}", symbol);
            } else {
                print!(" ");
            }
        }
        println!();
    }
}
